test/e2e/specs/updates/scaleoutin/scaleoutin.go

```diff
@@ -0,0 +1,216 @@
+package scaleoutin
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	apiappsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/fakerp"
+	"github.com/openshift/openshift-azure/pkg/util/randomstring"
+	"github.com/openshift/openshift-azure/pkg/util/ready"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+	"github.com/openshift/openshift-azure/test/clients/openshift"
+	"github.com/openshift/openshift-azure/test/e2e/specs/updates"
+)
+
+var _ = Describe("Scale Out/In E2E tests [ScaleOutIn] [Update]", func() {
+	defer GinkgoRecover()
+
+	var (
+		azurecli *azure.Client
+		occli    *openshift.Client
+		manifest = flag.String("manifest", "../../_data/manifest.yaml", "Path to the manifest to send to the RP")
+	)
+
+	const (
+		sampleDeployment = "hello-openshift"
+	)
+
+	BeforeEach(func() {
+		var err error
+		azurecli, err = azure.NewClientFromEnvironment()
+		Expect(err).NotTo(HaveOccurred())
+		occli, err = openshift.NewCustomerAdminClient()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should be possible to maintain a healthy cluster after scaling it out and in", func() {
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, api.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
+		ctx = context.WithValue(ctx, api.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
+		ctx = context.WithValue(ctx, api.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
+
+		By("Scaling out the compute scale set to 2 nodes")
+		By("Parsing the external manifest")
+		external, err := updates.ParseExternalConfig(*manifest)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(external).NotTo(BeNil())
+
+		By(fmt.Sprintf("Updating the size of the compute scale set to %d in the external manifest", 2))
+		Expect(external.Properties.AgentPoolProfiles[1].Name).To(Equal("compute"))
+		external.Properties.AgentPoolProfiles[1].Count = 2
+
+		By("Calling update on the rp with the updated external manifest")
+		pluginConfig, err := fakerp.GetPluginConfig()
+		Expect(err).NotTo(HaveOccurred())
+		updated, err := azurecli.UpdateOSACluster(ctx, external, pluginConfig)
```

Comment:
should be using a client library to be able to talk to either RP here.  Getting this set up neatly may take a bit of effort.

Reply:
i'm on it